app/components/document-creator.js

```diff
@@ -81,7 +81,7 @@ export default class DocumentCreatorComponent extends Component {
     if (this.template) {
       if (this.template.reload) {
         // regular templates from templatesForContext do not return body of template
-        await this.template.reload();
+        await this.template.reload(this.template);
```

Comment:
this is odd, reload doesn't require a param afaik:
https://api.emberjs.com/ember-data/3.28/classes/Model/methods/reload?anchor=reload

Reply:
aah but here's the fun part
template is not an ember-data model here
look at the RegulatoryAttachmentsFetcher service
it makes a raw sparql request and from the bindings it creates an object which _does_ have a reload method that takes an argument

Reply:
Ah yes sorry, forgot about this comment, basically I wanted to make it compatible with the old document creator and it was the best way I came up with

Reply:
If anyone has a better idea feel free to suggest it

Reply:
but, why? we already check if the object has a reload function and only execute reload then?

Reply:
Because I need a hook to fetch the template content, so I don't fetch all the contents before the user selects the one he wants to use.